hypothesis-python/src/hypothesis/_strategies.py

```diff
@@ -1292,13 +1292,17 @@ def from_type(thing):
             # At runtime, `typing.NewType` returns an identity function rather
             # than an actual type, but we can check that for a possible match
             # and then read the magic attribute to unwrap it.
-            if (
-                hasattr(thing, "__supertype__")
-                and hasattr(typing, "NewType")
-                and isfunction(thing)
-                and getattr(thing, "__module__", 0) == "typing"
-            ):
+            if types.is_a_new_type(thing):
+                # check if thing is explicitly registered strategy
```

Comment:
I'd expand this comment out, because `thing` is not a strategy but there may be a corresponding strategy registered.  Kinda pedantic, but it's the only way we can keep this logic straight. 

```python
# Check if we have an explicitly registered strategy for this thing,
# resolve it so, and otherwise resolve as for the base type.
```